llmfoundry/models/mpt/modeling_mpt.py

```diff
@@ -554,13 +560,12 @@ def forward(
         use_cache = (use_cache
                      if use_cache is not None else self.config.use_cache)
 
-        # if input_embeds is not none, raise a not implemented error
-        if inputs_embeds is not None:
-            raise NotImplementedError(
-                'inputs_embeds has to be None (for hf/peft support).')
-        # decoder outputs consists of (dec_features, layer_state, dec_hidden, dec_attn)
+        if input_ids is not None and inputs_embeds is not None:
+            raise ValueError(
+                'You cannot specify both input_ids and inputs_embeds.')
```

Comment:
isn't this already checked [here](https://github.com/mosaicml/llm-foundry/pull/687/files#diff-d99c63c3cfe678d1d5133c1a5f7cd670f20887c97add390e5cc021bc0a0b9127R365)